File: lib/iscp/messages/PhaseMatchingBassMsg.dart
/*
 * Enhanced Controller for Onkyo and Pioneer Pro
 * Copyright (C) 2019-2021 by Mikhail Kulesh
 *
 * This program is free software: you can redistribute it and/or modify it under the terms of the GNU
 * General Public License as published by the Free Software Foundation, either version 3 of the License,
 * or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
 * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details. You should have received a copy of the GNU General
 * Public License along with this program.
 */

import "../../constants/Strings.dart";
import "../EISCPMessage.dart";
import "EnumParameterMsg.dart";

enum PhaseMatchingBass
{
    NONE,
    OFF,
    ON,
    TOGGLE
}

/*
 * Phase Matching Bass Command
 */
class PhaseMatchingBassMsg extends EnumParameterMsg<PhaseMatchingBass>
{
    static const String CODE = "PMB";

    static const ExtEnum<PhaseMatchingBass> ValueEnum = ExtEnum<PhaseMatchingBass>([
        EnumItem.code(PhaseMatchingBass.NONE, "N/A",
            descrList: Strings.l_device_two_way_switch_none, defValue: true),
        EnumItem.code(PhaseMatchingBass.OFF, "00",
            descrList: Strings.l_device_two_way_switch_off),
        EnumItem.code(PhaseMatchingBass.ON, "01",
            descrList: Strings.l_device_two_way_switch_on),
        EnumItem.code(PhaseMatchingBass.TOGGLE, "TG",
            descrList: Strings.l_device_two_way_switch_toggle)
    ]);

    PhaseMatchingBassMsg(EISCPMessage raw) : super(CODE, raw, ValueEnum);

    PhaseMatchingBassMsg.output(PhaseMatchingBass v) : super.output(CODE, v, ValueEnum);

    @override
    bool hasImpactOnMediaList()
    {
        return false;
    }
}

File: app/src/main/java/com/mkulesh/onpc/iscp/messages/PhaseMatchingBassMsg.java
/*
 * Enhanced Music Controller
 * Copyright (C) 2018-2022 by Mikhail Kulesh
 *
 * This program is free software: you can redistribute it and/or modify it under the terms of the GNU
 * General Public License as published by the Free Software Foundation, either version 3 of the License,
 * or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
 * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details. You should have received a copy of the GNU General
 * Public License along with this program.
 */

package com.mkulesh.onpc.iscp.messages;

import com.mkulesh.onpc.R;
import com.mkulesh.onpc.iscp.EISCPMessage;
import com.mkulesh.onpc.iscp.ISCPMessage;

import androidx.annotation.NonNull;
import androidx.annotation.StringRes;

/*
 * Phase Matching Bass Command
 */
public class PhaseMatchingBassMsg extends ISCPMessage
{
    public final static String CODE = "PMB";

    public enum Status implements StringParameterIf
    {
        NONE("N/A", R.string.device_two_way_switch_none),
        OFF("00", R.string.device_two_way_switch_off),
        ON("01", R.string.device_two_way_switch_on),
        TOGGLE("TG", R.string.device_two_way_switch_toggle);

        final String code;

        @StringRes
        final int descriptionId;

        Status(final String code, @StringRes final int descriptionId)
        {
            this.code = code;
            this.descriptionId = descriptionId;
        }

        public String getCode()
        {
            return code;
        }

        @StringRes
        public int getDescriptionId()
        {
            return descriptionId;
        }
    }

    private final Status status;

    PhaseMatchingBassMsg(EISCPMessage raw) throws Exception
    {
        super(raw);
        status = (Status) searchParameter(data, Status.values(), Status.NONE);
    }

    public PhaseMatchingBassMsg(Status status)
    {
        super(0, null);
        this.status = status;
    }

    public Status getStatus()
    {
        return status;
    }

    @NonNull
    @Override
    public String toString()
    {
        return CODE + "[" + status.toString() + "]";
    }

    @Override
    public EISCPMessage getCmdMsg()
    {
        return new EISCPMessage(CODE, status.getCode());
    }

    @Override
    public boolean hasImpactOnMediaList()
    {
        return false;
    }
}
